### ISSUE TITLE
VersionedProperty support in the JSPs

### ISSUE THREAD

**Issue by eggrobin** [unicode-org/unicodetools#432]
The invariant test have a versioned property syntax, _e.g._, `\p{U4.1:XID_Continue}`, implemented in `VersionedProperty`.
https://github.com/unicode-org/unicodetools/blob/70dce2c89f185c65b436c28404ae5b7bdb32c2d1/unicodetools/src/main/java/org/unicode/text/UCD/TestUnicodeInvariants.java#L1033-L1072

Something like that would be useful for list-unicodeset.jsp.
It sounds like there is limited support for multiple versions with the β syntax, but I cannot figure out how that one is implemented.

**Comment by eggrobin** [comment]
> For the beta, the idea is that you have multiple release versions plus 1 dev version.

So, matching the subfolders of `ucd`? That was what I had in mind.

> Do you think we need multiple?

Multiple different βs? That sounds like it would confuse things rather than help.

**Comment by macchiati** [comment]
When I read "limited support for multiple versions with the β syntax", I
thought you were asking for multiple betas. Glad you're not!

**Comment by eggrobin** [comment]
> Should the invariants use it instead of ToolUnicodePropertySource?

Wait, they currently use _both_, with a preference for ToolUnicodePropertySource. So I guess the question is whether this should be inverted.

```java
         propSource = getProperties(version);  // ToolUnicodePropertySource.
         property = propSource.getProperty(xPropertyName); 
         if (property == null) { 
             propSource = getIndexedProperties(version);   // IndexUnicodeProperties.
             property = propSource.getProperty(xPropertyName); 
```

**Comment by macchiati** [comment]
I agree, it should be inverted. There are a few constructed properties
(non-UCD) that are only available via Tool... That is, the extra
normalization properties. So those should fallback to the Tool

**Comment by eggrobin** [comment]
> it should be inverted
Ah, that doesn’t work, because IndexUnicodeProperties, while closer to the truth for non-default values, it has issues with default values (I think it relies on @&zwnj;missing lines that are not always there).

At a glance, it is missing defaults at least for:
1. CCC, 
2. the non-special values of the properties in SpecialCasing (those used to have @&zwnj;missing lines, which IndexUnicodeProperties.txt knows how to read, but they got removed in 7.0), 
3.  the case foldings.

(The failings of IndexUnicodeProperties should probably turn into their own issue, but I am on a train with spotty internet, so I will be happy if this comment goes through.)

**Comment by macchiati** [comment]
> The failings of IndexUnicodeProperties should probably turn into their
own issue
Agreed, and thanks for tracking this down
